lib/vector-core/src/event/log_event.rs

```diff
@@ -150,7 +151,8 @@ impl LogEvent {
     /// valid for `LogNamespace::Legacy`
     pub fn from_str_legacy(msg: impl Into<String>) -> Self {
         let mut log = LogEvent::default();
-        log.insert(log_schema().message_key(), msg.into());
+        log.maybe_insert(PathPrefix::Event, log_schema().message_key(), msg.into());
+
         if let Some(timestamp_key) = log_schema().timestamp_key() {
             log.insert((PathPrefix::Event, timestamp_key), Utc::now());
         }
```

Comment:
For future reference: This pattern could now use the same `maybe_insert` helper, except that we would have to pass in `Utc::now()` unconditionally, which means grabbing system time even when there is no timestamp key. This is out of scope for the current PR but something to think about. You'd need to change `maybe_insert` to take a closure instead of an `Into`, so that adds complication.

Reply:
Yes, good point. There might be more cases like this with other LogSchema fields. My plan was/is to follow-up with a smaller PR because this one is getting too complex.